lektor/cli.py

```diff
@@ -285,14 +288,15 @@ def deploy_cmd(ctx, server, output_path, **credentials):
         server_info = config.get_server(server)
         if server_info is None:
             raise click.BadParameter('Server "%s" does not exist.' % server,
-                                     param_hint='server')
-
+                                     param_hint='server')  
     try:
         event_iter = publish(env, server_info.target, output_path,
-                             credentials=credentials, server_info=server_info)
+                             credentials=credentials, server_info=server_info, 
+                             message=message)
```

Comment:
You still need a message to cover this change. When the `deploy_cmd` is called with a message passed in on the command line, it needs to pass that message to the publisher. Here's how you might write a test like that:
```python
from click.testing import CliRunner

def test_deploy_message_proxies_to_publisher(tmpdir, mocker, env):
    mock_publisher = mocker.Mock(spec=Publisher)
    mocker.patch.dict(builtin_publishers, {'ghpages': mock_publisher})
    runner = CliRunner()

    runner.invoke(deploy_cmd, ['--message', 'a custom message'])

    mock_publish_method = mock_publisher.return_value.publish
    url = url_parse("rsync://myserver.com/path/to/website")
    mock_publish_method.assert_called_with(url, None, message='a custom message')
```

Let me break this down. The test starts by making a fake publisher class, which responds to all the same function calls as a real publisher class but does nothing. Then, we use `mocker.patch.dict` to slide this fake publisher into Lektor's dictionary of builtin publishers, to take the place of the `GithubPagesPublisher` class. Then we set up a Click `CliRunner` [as described in the Click documentation](http://click.pocoo.org/5/testing/#basic-testing), and use that to make the actual call that is being tested. That call should cause the publisher's `publish` method to be called, but since we've swapped out the real publisher for a fake one, it doesn't actually publish anything. Then we can assert that the fake publisher was actually called with that `message` that we passed on the command line. (The `url` variable is defined in the `tests/demo-project/Website.lektorproject` file.)

Does that make sense?